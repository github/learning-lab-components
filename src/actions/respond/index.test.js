const respond = require('.')
const mockContext = require('../../../tests/mockContext')

describe('respond', () => {
  let context

  beforeEach(() => {
    context = mockContext({
      issue: {
        number: 1
      }
    }, {
      issues: {
        createComment: jest.fn()
      },
      search: {
        issues: jest.fn(() => Promise.resolve({ data: { items: [{ title: 'An issue title', number: 2 }] } }))
      }
    })
  })

  it('responds to the issue in the payload', async () => {
    await respond(context, { with: 'a body' })
    expect(context.github.issues.createComment).toHaveBeenCalled()
    expect(context.github.issues.createComment.mock.calls).toMatchSnapshot()
  })

  it('responds to the issue with the specified number', async () => {
    await respond(context, { with: 'a body', issue: 2 })
    expect(context.github.issues.createComment).toHaveBeenCalled()
    expect(context.github.issues.createComment.mock.calls).toMatchSnapshot()
  })

  it('responds to the issue with the specified title', async () => {
    await respond(context, { with: 'a body', issue: 'An issue title' })
    expect(context.github.issues.createComment).toHaveBeenCalled()
    expect(context.github.issues.createComment.mock.calls).toMatchSnapshot()
  })
})
