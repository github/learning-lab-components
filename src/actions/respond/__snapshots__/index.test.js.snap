// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`respond responds to the issue in the payload 1`] = `
Array [
  Array [
    Object {
      "body": "a body",
      "number": 1,
      "owner": "JasonEtco",
      "repo": "example",
    },
  ],
]
`;

exports[`respond responds to the issue with the specified number 1`] = `
Array [
  Array [
    Object {
      "body": "a body",
      "number": 2,
      "owner": "JasonEtco",
      "repo": "example",
    },
  ],
]
`;

exports[`respond responds to the issue with the specified title 1`] = `
Array [
  Array [
    Object {
      "body": "a body",
      "number": 2,
      "owner": "JasonEtco",
      "repo": "example",
    },
  ],
]
`;
